lib/matplotlib/axis.py

```diff
@@ -53,7 +53,8 @@ class Tick(martist.Artist):
         The right/top tick label.
 
     """
-    def __init__(self, axes, loc, label,
+    @cbook._delete_parameter("3.3", "label")
+    def __init__(self, axes, loc, label=None,
```

Comment:
it's a bit strange because the Tick now takes a bunch of kwargs which basically get forwarded to the label Text objects (labelsize, labelcolor, labelrotation) but can't set the text itself?  (IOW should we make the label kwarg actually work instead?)  Alternatively should these labelfoos also get deprecated?

Reply:
I don't know. The label text is not really in the control of the tick but in the control of the Locator/Formatter. All usages in our codebase just passed an empty string. So making `label` work seems not too desirable.

I think label*foos* are currently used for creating additional ticks with the right properties after `tick_params()` was used. Would have to check for that.

Generally, I'd rather make the `Tick` API smaller than larger because it's a bit performance critical; and we may want to migrate to somthing like a TickCollection at some point.

Reply:
I strongly agree with making it smaller rather than larger, but would rather get rid of all label related args together if possible. Won't block over that though.

Reply:
I can look into that, but would rather do in a separate PR. I have the itch that everything will break down if I remove the label*foos*. And sorting that out in a small clean PR is simpler. Also it's easier to revert if we find out this was a bad idea.